@ExplicitParameterInjection is now by default inherited for nested classes

manovotn opened the pull request:
Fixes #164 

`@ExplicitParameterInjection` now has a configurable `boolean` value indicating whether Weld should apply this behavior.
Nested classes are automatically inspecting their enclosing classes for this annotation and it's value unless they declare it themselves.

mkouba commented on junit5/src/main/java/org/jboss/weld/junit5/ExplicitParamInjection.java:
```diff
@@ -22,16 +22,26 @@
+     *
+     * @return {@code true} by default; can be explicitly set to {@code false} to make Weld attempt to resolve all parameters
+     */
+    boolean apply() default true;
```
Hm, why don't we use `value()`? This would allow for `@ExplicitParamInjection(false)` which is IMO more convenient.

manovotn replied:
Ah, this is a leftover from the previous attempt; will do!

mkouba commented on junit5/src/test/java/org/jboss/weld/junit5/explicitInjection/ExplicitParameterInjectionNestedClass2Test.java:
```diff
@@ -0,0 +1,98 @@
+
+    @Test
+    public void testParameterResolution(@Default Foo foo, Bar bar, @MyQualifier BeanWithQualifier bean) {
+        // Bar should be resolved by another extension
```
In this test, Bar is also resolved by Weld, or?

manovotn replied:
Yes, the comment below is a misleading a copy-paste error meant to confuse any reviews :D

manovotn commented:
Edited README, fixed copy-paste errors and tests and added a test for method-level declaration of this annotation which can now technically say `@ExplicitParameterInjection(false)` even though I cannot imagine why you'd do that.
